addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjectionTimingScanRule.java

```diff
@@ -0,0 +1,269 @@
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractAppParamPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.addon.commonlib.CommonAlertTag;
+import org.zaproxy.addon.commonlib.timing.TimingUtils;
+import org.zaproxy.zap.extension.ruleconfig.RuleConfigParam;
+import org.zaproxy.zap.model.Tech;
+import org.zaproxy.zap.model.TechSet;
+
+/**
+ * The MongoInjection scan rule identifies MongoDB injection vulnerabilities
+ *
+ * @author l.casciaro
+ */
+public class MongoDbInjectionTimingScanRule extends AbstractAppParamPlugin
+        implements CommonActiveScanRuleInfo {
+
+    // Prefix for internationalised messages used by this rule
+    private static final String MESSAGE_PREFIX = "ascanalpha.mongodb.";
+
+    private static final String RULE_SLEEP_TIME = RuleConfigParam.RULE_COMMON_SLEEP_TIME;
+
+    private static final int BLIND_REQUEST_LIMIT = 4;
+    private static final int DEFAULT_TIME_SLEEP_SEC = 5;
+    private static final double TIME_CORRELATION_ERROR_RANGE = 0.15;
+    private static final double TIME_SLOPE_ERROR_RANGE = 0.30;
+
+    private static final String SLEEP_ATTACK = "sleep";
+
+    private static final List<String> SLEEP_INJECTION =
+            List.of(
+                    // Attacking $where clause
+                    // function() { var x = md5( some_input ); return this.password == x;}
+                    "\"); sleep({0}); print(\"",
+                    "'; sleep({0}); print(\'",
+                    "0); sleep({0}); print(\"",
+                    // function() { var x = this.value == some_input; return x;}
+                    "'; sleep({0}); var x='",
+                    "\"; sleep({0}); var x=\"",
+                    "0; sleep({0})",
+                    // function() { return this.value == some_input }
+                    "zap' || sleep({0}) && 'zap'=='zap",
+                    "zap\" || sleep({0}) && \"zap\"==\"zap",
+                    "0 || sleep({0})",
+                    // function() { return this.value == some_function(some_imput) }
+                    "zap') || sleep({0}) && hex_md5('zap",
+                    "zap\") || sleep({0}) && md5(\"zap",
+                    "0)  || sleep({0})",
+                    // Attacking mapReduce clause (only for old versions of mongodb)
+                    "_id);}, function(inj) { sleep({0});return 1;}, { out: 'x'}); db.injection.mapReduce(function() { emit(1,1");
+
+    private static final Logger LOGGER = LogManager.getLogger(MongoDbInjectionTimingScanRule.class);
+    private static final Map<String, String> ALERT_TAGS =
+            CommonAlertTag.toMap(
+                    CommonAlertTag.OWASP_2021_A03_INJECTION,
+                    CommonAlertTag.OWASP_2017_A01_INJECTION,
+                    CommonAlertTag.WSTG_V42_INPV_05_SQLI);
```

Comment:
I think a `CommonAlertTag.TEST_TIMING` or similar would be _really_ useful.
Esp if/when (hopefully when) we support tag based policy rules, eg "inc: ["OWASP_2021_*"], exc: ["TEST_TIMING"]` if you see what I mean...